.github/workflows/ci.yml

```diff
@@ -505,8 +549,21 @@ jobs:
           name=$(echo -n "${{ matrix.modules }}" | sed -e 's/[":<>|\*\?\\\/]/_/g')
           echo "ARTIFACT_NAME=$name" >> $GITHUB_ENV
       - uses: ./.github/actions/upload
+        # Always upload the test report for the annotate.yml workflow, but only the single XML file to keep the artifact small
+        if: always()
         with:
           name-suffix: ${{ env.ARTIFACT_NAME }}
+          # Upload all test reports only on failure, because the artifacts are large
+          upload-results: >-
+            ${{ steps.test.conclusion == 'failure' ||
```

Comment:
I don't like how verbose this is and I'm starting to think moving uploads to a composite action was a mistake. @hashhar WDYT?

Reply:
I'll close this in favor of #15181